`bee-rest-api` 0.1.6 compilation fails

## Bug description

The `bee-rest-api` 0.1.6 crate fails to compile with the latest changes in `bee-common` 0.5.0, causing `bee-ledger` 0.5.0 to fail compilation (errors at bottom of issue).

Not sure exactly why there's a conflict, maybe because two versions of `bee-common` are being pulled into the dependency tree?

Edit: I think it's because `bee-message` had its patch version incremented (0.1.6) so it pulls in the later crate versions automatically?

```
> cargo tree -i bee-common:0.4.1
bee-common v0.4.1
└── bee-ledger v0.5.0
    └── bee-rest-api v0.1.6
        └── testbed v0.1.0 (C:\Work\iota\testbed)                                                                                                                                                                           

> cargo tree -i bee-common:0.5.0
bee-common v0.5.0
└── bee-message v0.1.6
    ├── bee-ledger v0.5.0
    │   └── bee-rest-api v0.1.6
    │       └── testbed v0.1.0 (C:\Work\iota\testbed)
    ├── bee-protocol v0.1.1
    │   └── bee-rest-api v0.1.6 (*)
    └── bee-rest-api v0.1.6 (*) 
```

## Rust version

Which version of Rust are you running?

- Rust version: 1.57.0 (2021-10-21)

## Bee version

Which version of Bee are you using?

- Bee version (version number, commit, or branch): `bee-rest-api` 0.1.6

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.

1. Add `bee-rest-api` 0.1.6 as a dependency. `cargo add bee-rest-api`
2. `cargo check`
3. Compilation fails.


E.g. Cargo.toml with no other dependencies.
```
[package]
name = "testbed"
version = "0.1.0"
edition = "2021"

[dependencies]
bee-rest-api = "0.1.6"    
```

## Expected behaviour

Describe what you expect to happen.

Compiles successfully.

## Actual behaviour

Describe what actually happens.

The `bee-ledger` 0.5.0 dependency fails to compile with errors that `unpack_inner` is not recognised.

## Errors

Paste any errors that you see, including logs, errors, or screenshots.

```
    Checking bee-ledger v0.5.0
error[E0599]: no function or associated item named `unpack_inner` found for struct `MilestoneIndex` in the current scope
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-ledger-0.5.0\src\types\snapshot\header.rs:81:41
   |
81 |         let sep_index = MilestoneIndex::unpack_inner::<R, CHECK>(reader)?;
   |                                         ^^^^^^^^^^^^ function or associated item not found in `MilestoneIndex`
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
4  | use bee_common::packable::Packable;
   |

error[E0599]: no function or associated item named `unpack_inner` found for struct `MilestoneIndex` in the current scope
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-ledger-0.5.0\src\types\snapshot\header.rs:82:44
   |
82 |         let ledger_index = MilestoneIndex::unpack_inner::<R, CHECK>(reader)?;
   |                                            ^^^^^^^^^^^^ function or associated item not found in `MilestoneIndex`
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
4  | use bee_common::packable::Packable;
   |

error[E0599]: no method named `packed_len` found for struct `MilestoneId` in the current scope
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-ledger-0.5.0\src\types\snapshot\header.rs:138:49
    |
138 |             + self.treasury_output_milestone_id.packed_len()
    |                                                 ^^^^^^^^^^ method not found in `MilestoneId`
    |
   ::: C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-common-0.5.0\src\packable.rs:14:8
    |
14  |     fn packed_len(&self) -> usize;
    |        ---------- the method is available for `MilestoneId` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
4   | use bee_common::packable::Packable;
    |

<snip, all errors are similar>

For more information about this error, try `rustc --explain E0599`.
error: could not compile `bee-ledger` due to 71 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed    
```

--- a/bee-api/bee-rest-api/CHANGELOG.md
+++ b/bee-api/bee-rest-api/CHANGELOG.md
@@ -19,6 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 0.1.7 - 2021-12-08
+
+### Changed
+
+- Update `bee-ledger` dependency;
+
 ## 0.1.6 - 2021-12-06
 
 ### Fixed

--- a/bee-api/bee-rest-api/Cargo.toml
+++ b/bee-api/bee-rest-api/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "bee-rest-api"
-version = "0.1.6"
+version = "0.1.7"
 authors = [ "IOTA Stiftung" ]
 edition = "2021"
 description = "The default REST API implementation for the IOTA Bee node software."

